src/Development/IDE/Core/PositionMapping.hs

```diff
@@ -32,53 +32,27 @@ idMapping :: PositionMapping
 idMapping = PositionMapping Just Just
 
 applyChange :: PositionMapping -> TextDocumentContentChangeEvent -> PositionMapping
-applyChange posMapping (TextDocumentContentChangeEvent (Just r) _ t) = PositionMapping
-    { toCurrentPosition = toCurrent r t <=< toCurrentPosition posMapping
-    , fromCurrentPosition = fromCurrentPosition posMapping <=< fromCurrent r t
+applyChange posMapping (TextDocumentContentChangeEvent (Just (Range start end)) _ t) = PositionMapping
+    { toCurrentPosition = morph start end (newEnd t start) <=< toCurrentPosition posMapping
```

Comment:
Probably easier to read if you keep `fromCurrent` and `toCurrent` wrappers for `morph` and then the tests should also compile.